Sources/ParseSwift/Authentication/3rd Party/ParseFacebook.swift

```diff
@@ -0,0 +1,395 @@
+        /// - returns: `true` if all the mandatory keys are present, `false` otherwise.
+        func verifyMandatoryKeys(authData: [String: String]?) -> Bool {
+            guard let authData = authData,
+                  authData[AuthenticationKeys.id.rawValue] != nil,
+                  authData[AuthenticationKeys.expirationDate.rawValue] != nil else {
+                return false
+            }
+            
+            if let _ = authData[AuthenticationKeys.accessToken.rawValue] {
+                return true
+            }else if let _ = authData[AuthenticationKeys.authenticationToken.rawValue] {
+                return true
+            }
+            return false
+        }
+    }
+    public static var __type: String { // swiftlint:disable:this identifier_name
+        "facebook"
+    }
+    public init() { }
+}
+
+// MARK: Login
+public extension ParseFacebook {
+    /**
+     Login a `ParseUser` *asynchronously* using Facebook authentication for limited login.
+     - parameter userId: The `Facebook userId` from `FacebookSDK`.
+     - parameter authenticationToken: The `authenticationToken` from `FacebookSDK`.
+     - parameter options: A set of header options sent to the server. Defaults to an empty set.
+     - parameter callbackQueue: The queue to return to after completion. Default value of .main.
+     - parameter completion: The block to execute.
+     */
+    func login(userId: String,
+               authenticationToken: String?,
+               expirationDate: Date,
+               options: API.Options = [],
+               callbackQueue: DispatchQueue = .main,
+               completion: @escaping (Result<AuthenticatedUser, ParseError>) -> Void) {
+
+        guard let facebookAuthData = try? AuthenticationKeys.id.makeDictionary(userId: userId, accessToken: nil,
+                                                                          authenticationToken: authenticationToken, expirationDate: expirationDate) else {
+            callbackQueue.async {
+                completion(.failure(.init(code: .unknownError,
+                                          message: "Couldn't create authData.")))
+            }
+            return
+        }
+        login(authData: facebookAuthData,
+              options: options,
+              callbackQueue: callbackQueue,
+              completion: completion)
+    }
+    /**
+     Login a `ParseUser` *asynchronously* using Facebook authentication for graph API login.
+     - parameter userId: The `Facebook userId` from `FacebookSDK`.
+     - parameter accessToken: The `accessToken` from `FacebookSDK`.
+     - parameter options: A set of header options sent to the server. Defaults to an empty set.
+     - parameter callbackQueue: The queue to return to after completion. Default value of .main.
+     - parameter completion: The block to execute.
+     */
+    func login(userId: String,
+               accessToken: String?,
```

Comment:
```suggestion
               accessToken: String,
```

Reply:
If a developer is using this method it is “mandatory” they specify `accessToken`. I made this change in two places, but you should make the change in the rest.